pkgs/development/interpreters/python/pypy/2.7/default.nix

```diff
@@ -23,22 +23,20 @@ in stdenv.mkDerivation rec {
 
     src = fetchurl {
       url = "https://bitbucket.org/pypy/pypy/get/release-pypy${pythonVersion}-v${version}.tar.bz2";
-      sha256 = "145a0kd5c0s1v2rpavw9ihncfb05s2x7chc70v8fssvyxq601911";
+      sha256 = "0dibf1bx4icrbi8zsqk7cfwgwsd3hfx6biz59k8j5rys3fx9z418";
     };
 
-   # http://bugs.python.org/issue27369
-    postPatch = let
-      expatch = fetchurl {
-        name = "tests-expat-2.2.0.patch";
-        url = "http://bugs.python.org/file43514/0001-Fix-Python-2.7.11-tests-for-Expat-2.2.0.patch";
-        sha256 = "1j3pa7ly9xrhp8jjwg5l77z7i3y68gx8f8jchqk6zc39d9glq3il";
-      };
-      in ''
-      patch lib-python/2.7/test/test_pyexpat.py < '${expatch}'
+    patches = [
+      # https://bitbucket.org/pypy/pypy/issues/2604/lib-python-27-test-test_ospy
+      ./2604-skip-urandom-fd-test.patch
+    ];
+
+    postPatch = ''
       substituteInPlace "lib-python/2.7/lib-tk/Tix.py" --replace "os.environ.get('TIX_LIBRARY')" "os.environ.get('TIX_LIBRARY') or '${tix}/lib'"
     '';
 
     buildInputs = [ bzip2 openssl pkgconfig python libffi ncurses expat sqlite tk tcl xlibsWrapper libX11 makeWrapper gdbm db ]
+      ++ (with pythonPackages; [ pycparser ])
```

Comment:
Maybe its good to use `python.withPackages` here so that its clear that `pycparser` is for that instance of Python.